cytotable/convert.py

```diff
@@ -377,14 +347,9 @@ def _source_chunk_to_parquet(
                 table=ddb_reader.execute(
                     f"""
                     {base_query}
+                    WHERE {source['page_key']} BETWEEN {pageset[0]} AND {pageset[1]}
                     /* order by all columns for deterministic output */
-                    ORDER BY ALL
-                    LIMIT {chunk_size} OFFSET {offset}
-                    """
-                    if sort_output
-                    else f"""
-                    {base_query}
-                    LIMIT {chunk_size} OFFSET {offset}
+                    {"ORDER BY " + source['page_key'] if sort_output else ""};
```

Comment:
I'm likely not reading this correctly, but is this ordering by `page_key` for every chunked/pageset operation? Wouldn't it be more efficient to order after all pagesets are processed?

Reply:
Thanks, great thought - this was a vestige of the earlier offser-based pagination and I believe we could update here for more performant operations as you mentioned.

Reply:
After looking into this more I felt that it'd be best to leave this line as-is and add additional changes to ensure the sorted results arrive sorted within the concatenation phase of CytoTable. I opted to do this because sorting for individual pages vs entirety of the dataset will have a lower performance impact and be more likely to succeed alongside resource constraints.

The result here will sort each pageset meaning the results within a single range of values are sorted. Afterwards, during concatenation to a single file, the results are combined using a natural sort over the filenames (which include the pageset range) to create a fully sorted dataset per source table and eventually joined table.